pymc/distributions/timeseries.py

```diff
@@ -157,15 +157,28 @@ def dist(
 
         mu = at.as_tensor_variable(floatX(mu))
         sigma = at.as_tensor_variable(floatX(sigma))
-        if steps is None:
-            raise ValueError("Must specify steps parameter")
+        shape = kwargs.get("shape", None)
+
+        # Get number of steps directly or from shape
+        if steps is not None and shape is not None:
+            raise ValueError("Specify only one of steps or shape parameter")
```

Comment:
This ValueError is wrong, shape contains more information than steps, including batched dimensions as in `shape=(4, 2)`

Reply:
I agree that shape contains more information. What I'm catching in this error is if someone specifies both, for example
What happens if someone specifies something like `steps=1, shape=(2,5)` Which one is correct?

Or are you saying that this value error should just check if someone specifies a mismatch for example
`steps=4, shape=(2,5)` should pass?

Reply:
You can add an assert to the step argument if you're worried about that.

```python
steps = Assert("steps do not match last shape dimension")(shape[-1], at.eq(shape[-1], steps))
```

But I don't think it's a big problem to just overwrite it silently to be honest.

Reply:
Asserts shoudlnt be used for production enforcement so an exception of some sort is the correct here. I can update the value error.

I dont think we should silently override it. It can be confusing for a user who doesnt have the indepth knowledge of internals like we do

Reply:
That assert is a Symbolic exception, not a Python assert. The problem is that you cannot check what you want to check when the user calls the class, because both shapes and steps can be symbolic.

Also Aesara asserts can raise specific types of Errors, including `ValueError`: https://github.com/aesara-devs/aeppl/blob/0624b6b9b3301a81a74f5cf98c775c3f65b2f285/aeppl/logprob.py#L15-L29

Reply:
Oh man you are absolutely right. I keep forgetting these inputs can be symbolic and not actually determined till later. Thank you for pointing this out. Ill make the change in the next 96 hours

Reply:
Ugh... just realized that `shape` may have an `ellipsis`, which we didn't account for when trying to resize the `init` distribution... Allowing shape/ellipsis in `dist` open all sort of problems for composite distributions like this

Reply:
Updated to what I think is right